src/lab10.py

```diff
@@ -10,22 +10,14 @@
 import tkinter.font
 import urllib.parse
 import dukpy
-from lab3 import get_font
-from lab4 import print_tree
-from lab4 import Element
-from lab4 import Text
-from lab4 import HTMLParser
-from lab5 import DrawRect
-from lab6 import cascade_priority
-from lab6 import resolve_url
-from lab6 import style
-from lab6 import tree_to_list
-from lab6 import CSSParser
+from lab2 import WIDTH, HEIGHT, HSTEP, VSTEP, SCROLL_STEP
```

Comment:
Why do you need to change the imports in some of the labs? Were they wrong before?

Reply:
They were. This is a dumb thing that it would be good to automate, but right now the outline only includes things you explicitly `from ... import`. So for example, in one of the labs we actually call `DocumentLayout` but that in turn calls `BlockLayout` etc. If we only `from ... import DocumentLayout`, the outline will have `DocumentLayout` but not `BlockLayout`. So you have to import all of the transitive dependencies as well. Ideally I would construct use-def chains in the compiler so this could be automated but I haven't done that yet.